otri/database/database_stream.py

```diff
@@ -52,25 +78,38 @@ def __next__(self):
                 When no result is given by the database (the query is over),
                 closes the cursor and raises StopIteration again.
         '''
-        try:
-            return next(self.__cursor)
-        except StopIteration:
-            self.__cursor.close()
+        if self.__cursor.closed:
             raise StopIteration
+        if self.__buffer != None:
+            item = self.__buffer
+            self.__buffer = None
+            return item
+        else:
+            return next(self.__cursor)
 
     def has_next(self) -> bool:
         '''
         Returns:
             True if the cursor is still open (there are still rows to retrieve)
```

Comment:
True if there's another item, false otherwise. Let's remove the dependency from the cursor to the outside environment.